clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1476,7 +1481,10 @@ public void unsubscribe() {
                     subscriptions.assignedPartitions());
 
             try {
-                processBackgroundEvents(unsubscribeEvent.future(), timer);
+                // If users subscribe to an invalid topic name, they will get InvalidTopicException in error events,
+                // because network thread keeps trying to send MetadataRequest in the background.
+                // Ignore it to avoid unsubscribe failed.
+                processBackgroundEvents(unsubscribeEvent.future(), timer, e -> e instanceof InvalidTopicException);
```

Comment:
If unsubscribing from a topic that was never subscribed to is valid, should we also swallow the `TopicAuthorizationException`?

Reply:
Create a followup Jira https://issues.apache.org/jira/browse/KAFKA-17648.